Orange/widgets/data/owpreprocess.py

```diff
@@ -535,6 +536,125 @@ def createinstance(params):
 # TODO: Model based FS (random forest variable importance, ...), RFE
 # Unsupervised (min variance, constant, ...)??
 
+class RandomFeatureSelectEditor(BaseEditor):
+    #: Strategy
+    Fixed, Percentage, FDR, FPR, FWE = 0, 1, 2, 3, 4
+
+    def __init__(self):
+        super().__init__()
+        self._spins = {}
+        self._max_num_attributes = None
+        self._ticks_to_update_slider = [None, None]
+        self._strategy = RandomFeatureSelectEditor.Fixed
+        self._k = 10
+        self._p = 75
+
+        self.setLayout(QVBoxLayout())
+        box = gui.widgetBox(widget=self, box="Strategy", spacing=8)
+        strategy_buttons = gui.radioButtons(widget=box, master=self,
+                                            value="_strategy",
+                                            callback=self.update_strategy)
+
+        fixed_rb = gui.appendRadioButton(group=strategy_buttons, label="Fixed:",
+                                         insertInto=box)
+        fixed_rb.setMinimumHeight(25)
+        ibox = gui.indentedBox(widget=fixed_rb, sep=63)
+        k_spin = gui.spin(
+            widget=ibox, master=self, value="_k", minv=1, maxv=10, step=1,
+            label="", controlWidth=60, spinType=int, callback=self.edited.emit,
+            disabled=self._strategy != RandomFeatureSelectEditor.Fixed)
+        self._spins[RandomFeatureSelectEditor.Fixed] = k_spin
+
+        percentage_rb = gui.appendRadioButton(group=strategy_buttons,
+                                              label="Percentage:",
+                                              insertInto=box)
+        ibox = gui.indentedBox(widget=box)
+        p_spin = gui.hSlider(
+            widget=ibox, master=self, value="_p", minValue=0, maxValue=100,
+            step=4, ticks=20, labelFormat="%d %%", callback=self.update_p_spin,
+            disabled=self._strategy != RandomFeatureSelectEditor.Percentage)
+        self._spins[RandomFeatureSelectEditor.Percentage] = p_spin
+
+    def update_strategy(self, edited=True):
+        for strategy, spin in self._spins.items():
+            if self._strategy == strategy:
+                spin.setDisabled(False)
+            else:
+                spin.setDisabled(True)
+
+        if edited:
+            self.edited.emit()
+        else:
+            self.changed.emit()
+
+    def update_p_spin(self):
+        curr_val = self._p
+        prev_val = self._ticks_to_update_slider[1]
+        if (self._ticks_to_update_slider[0] is not None and (prev_val is None or
+            any([curr_val < tick <= prev_val or curr_val >= tick > prev_val
```

Comment:
`any()` is great, but it also accepts a generator and returns as soon as it finds the first truish item. No need to build the whole list.